pkg/clientwrappers/parametercontext/parametercontext.go

```diff
@@ -365,3 +375,16 @@ func updateRequest2Status(updateRequest *nigoapi.ParameterContextUpdateRequestEn
 		NotFoundRetryCount: 0,
 	}
 }
+
+func extractMostRecentSecretUpdateTime(parameterSecrets []*corev1.Secret) *metav1.Time {
+	var maximumSecretTime *metav1.Time = nil
+	for _, secret := range parameterSecrets {
+		for _, managedField := range secret.ObjectMeta.ManagedFields {
```

Comment:
Do you think we should filter by `managedField.operation == "Update"` here to capture when the last actual update was? Or do you want all operations (Apply, Update, etc)?

Reply:
I chose not to filter, thinking that there was no point in doing so. For me, whatever the action on the `Secret`, we should update the parameter because why not. I don't think it will cause spamming. But may be, I'm wrong?

Reply:
I don't think it should spam. I'm looking at secrets managed by Argo and i it doesn't spam the managedFields, so it should be okay.